src/main/java/exnihilocreatio/util/Util.java

```diff
@@ -87,10 +88,7 @@ public static void dropItemInWorld(TileEntity source, EntityPlayer player, ItemS
     }
 
     @SideOnly(Side.CLIENT)
-    public static TextureAtlasSprite getTextureFromBlockState(IBlockState state) {
-        if (state == null)
```

Comment:
Why? one is game breaking one is not? @NonNull just crashes the game when it encounters Null, doesn't gracefully just skip it like the if check.

Reply:
I mostly did it as no BlockState should be null and it feels like a bit of a performance waste to always check for null when we should expect it not to be null, plus anyone linking into that code will be warned when sending a null or potentially null BlockState . It also helps us in the event of a null BlockState track down where it's happening

Reply:
your @NonNull adds a null check, so that performance part is completely irrelevant, shouldn't happen, but might